src/components/Hero.js

```diff
@@ -4,5 +4,16 @@ export default {
 	props: ["error"],
 	components: {
 		Filedrop
+	},
+	methods: {
+		toggleModal() {
+			// Quick and clean way to emit to grandparent
+			// that modal should be opened/closed
+			let vm = this;
+			while (vm) {
```

Comment:
I'm not super deep into Vue, but is this command practise? It looks really hacky.

Don't events bubble in Vue?

Reply:
Events don't bubble to parent elements, unless you do it yourself. This is a way to bubble up the tree all the way to the über-grandparent. I believe the alternative would be to have `v-on` listeners in every child between the one emitting the event, and the one receiving the event. Although the current solution looks hacky, I can live with it :-)

Reply:
Hm I see: https://stackoverflow.com/questions/41993508/vuejs-bubbling-custom-events

Let's stick to this for now. If more of these situations arise we could look into an event bus.

Reply:
Looks like I'm doing option two, as I feel it is less cumbersome as option one (the `v-on` listeners).